### PR TITLE
FEATURE: CKE5 placeholder insert plugin

### PR DESCRIPTION
**Pull request by dimaip**
Requires: https://github.com/neos/neos-ui/pull/2546

Adds a dropdown to CKE5 that allows you to insert Fluid placeholders for elements configured in the current form. It takes either `formElement.properties.identifier` or `formElement.identifier` as the placeholder value.

Sample config to enable this feature:

```yaml
'Neos.Form.Builder:ConfirmationFinisher':
  properties:
    'message':
      ui:
        inspector:
          editor: 'Neos.Neos/Inspector/Editors/RichTextEditor'
          editorOptions:
            formatting:
              placeholderInsert: true
              strong: true
              em: true
```

image (2)

### CODE REVIEW

File: Resources/Private/PlaceholderInsert/src/PlaceholderInsertDropdown.js [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+          node.properties.label || node.properties.identifier || node.identifier
+      }));
+
+    if (options.length === 0) {
+      return null;
+    }
+
+    const placeholderLabel = this.props.i18nRegistry.translate(
+      "Neos.Form.Builder:Main:placeholder",
+      "Insert placeholder"
```
Comment (lorenzulrich) [inline Resources/Private/PlaceholderInsert/src/PlaceholderInsertDropdown.js]: This should be localized.

Comment (dimaip) [reply Resources/Private/PlaceholderInsert/src/PlaceholderInsertDropdown.js]: It is, once the translations will be in place, it would show up, this is just a fallback

File: Resources/Private/PlaceholderInsert/src/PlaceholderInsertDropdown.js [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+    const elementsPath = `${formPath}/elements@${workspace}`;
+
+    const elementsNode = this.props.nodesByContextPath[elementsPath];
+    if (!elementsNode) {
+      return null;
+    }
+    const options = elementsNode.children
+      .map(node => this.props.nodesByContextPath[node.contextPath])
+      .map(node => ({
+        value: node.properties.identifier || node.identifier,
```
Comment (lorenzulrich) [inline Resources/Private/PlaceholderInsert/src/PlaceholderInsertDropdown.js]: Would it be possible to have this in `Settings`? While thinking of it, I see that different use-cases are possible; sometimes you want to use speaking identifiers, sometimes not.

Comment (dimaip) [reply Resources/Private/PlaceholderInsert/src/PlaceholderInsertDropdown.js]: Could you describe your idea in more details: setting name, setting format, what exactly it does etc?

File: Resources/Private/PlaceholderInsert/package.json [hunk 3]
```diff
@@ -0,0 +1,15 @@
+{
+    "description": "PlaceholderInsert",
+    "license": "MIT",
+    "private": true,
+    "scripts": {
+        "build": "neos-react-scripts build",
+        "watch": "neos-react-scripts watch"
+    },
+    "devDependencies": {
+        "@neos-project/neos-ui-extensibility": "^1.3"
```
Comment (markusguenther) [inline Resources/Private/PlaceholderInsert/package.json]: really version 1.3?

**Comment by lorenzulrich** [comment]
We need to rethink if and when to add this to Form Builder:

* Form Builder currently supports Neos 3.1+ and the old UI. 
* As far as I know, the old UI is also supported in Neos 4.3, even though I'm not sure if anyone still uses it. If we can remove support for the old UI, we could have a release targeting Neos 4.3+ without support for the old UI.
* The underlying change in the Neos UI, neos/neos-ui#2546, is only targeting Neos 5. So we would need to have a separate version/branch with Neos 5 only support.

File: Resources/Private/PlaceholderInsert/src/PlaceholderInsertDropdown.js (hunk 2 above)
Comment (lorenzulrich) [reply Resources/Private/PlaceholderInsert/src/PlaceholderInsertDropdown.js]: After thinking again, let's not over-complicate things for the moment and go with it as it is.

File: Resources/Private/PlaceholderInsert/package.json (hunk 3 above)
Comment (dimaip) [reply Resources/Private/PlaceholderInsert/package.json]: It doesn't matter really, it would compile with any version of the extensibility package. Will change to wildcard to avoid confusion.

**Comment by lorenzulrich** [comment]
@dimaip just pointed out that this will not break anything even in older versions because it only works if the new UI is enabled and the UI has a `RichTextEditor`. Therefore, we could merge this without any drawbacks.

**Comment by paavo** [comment]
Thanks very much @dimaip for this awesome Feature. 
**Is there any hint, how to make this work with the `Neos.Form.Builder:EmailFinisher` too?**
Looks like the "Frontend Part" works well, but the Values are not available when sending the E-Mail 😢 

The "missing Property Selector" is still a reason, most of our Editors are not able to create their own Forms. 
Editors are not able to edit HTML in the ConfirmationFinisher / EmailFinisher. 
If this works, im pretty sure our Editors would be very happy 🎉

**Comment by dimaip** [comment]
Ohh @paavo sorry I totally forgot everything I did for this feature, so can't answer it from top of my head.
Maybe somebody who actually uses this feature could help? @lorenzulrich? @bwaidelich?

**Comment by bwaidelich** [comment]
> Maybe somebody who actually uses this feature could help?

I don't use this feature to be honest, but I tried to help in the #guild-form-builder channel: https://neos-project.slack.com/archives/CH86S151D

*TL;DR* This change prepares the `RichTextEditor` for Confirmation and EmailFinisher but the latter expects form value placeholders in the form `formValues.*` so that can't work. https://github.com/neos/form/pull/122 solves this by making the prefix optional..


BTW: The prefix was there to prevent naming conflicts with options for the finisher. An alternative solution might be to encourage the same prefix in the `ConfirmationFinisher` and to extend the `RichTextEditor` by some "placeholderPrefix" option
